targets/switch-test/switch_test.cpp

```diff
@@ -0,0 +1,121 @@
+/**
+ * This code will allow for an easy testing of the APM MOSFET switches
+ */
+
+#include <EVT/io/manager.hpp>
+#include <EVT/io/pin.hpp>
+#include <EVT/io/UART.hpp>
+#include <APM/APMUart.hpp>
+#include <cstring>
+#include <APM/APMManager.hpp>
+#include <string>
+
+#pragma clang diagnostic push
+#pragma ide diagnostic ignored "EndlessLoop"
+
+namespace IO = EVT::core::IO;
+
+namespace APM {
+
+constexpr int BAUD_RATE = 115200;
+constexpr int BUF_SIZE = 256;
+
+#ifdef NUCLEO_COMPILATION
+constexpr IO::Pin UART_TX = IO::Pin::UART_TX;
+constexpr IO::Pin UART_RX = IO::Pin::UART_RX;
+#else
+constexpr IO::Pin UART_TX = IO::Pin::PB_10;
+constexpr IO::Pin UART_RX = IO::Pin::PB_11;
+#endif
+
+char buf[BUF_SIZE];
+
+}  // namespace APM
+
+/**
+ * Returns the inverse of the current GPIO state
+ * @param state the current state of a GPIO pin
+ * @return the opposite state
+ */
+EVT::core::IO::GPIO::State get_toggle(EVT::core::IO::GPIO::State state) {
+    if (state == EVT::core::IO::GPIO::State::LOW) {
+        return EVT::core::IO::GPIO::State::HIGH;
+    } else {
+        return EVT::core::IO::GPIO::State::LOW;
+    }
+}
+
+int main() {
+    // Initialize IO Objects
+    IO::init();
+    IO::UART &uart = IO::getUART<APM::UART_TX, APM::UART_RX>(APM::BAUD_RATE);
+    IO::GPIO &accessorySW_GPIO =
+            IO::getGPIO<APM::APMManager::ACCESSORY_SW>(IO::GPIO::Direction::OUTPUT);
+    IO::GPIO &chargeSW_GPIO =
+            IO::getGPIO<APM::APMManager::CHARGE_SW>(IO::GPIO::Direction::OUTPUT);
+    IO::GPIO &chargeEnable_GPIO =
+            IO::getGPIO<APM::APMManager::CHARGE_LIMITER_ENABLE>(IO::GPIO::Direction::OUTPUT);
+    IO::GPIO &vicorSW_GPIO =
+            IO::getGPIO<APM::APMManager::VICOR_SW>(IO::GPIO::Direction::OUTPUT);
+    IO::GPIO &accessoryIndicator_GPIO =
+            IO::getGPIO<APM::APMManager::ACCESSORY_INDICATOR>(IO::GPIO::Direction::OUTPUT);
+    IO::GPIO &chargeIndicator_GPIO =
+            IO::getGPIO<APM::APMManager::Test_LED_0>(EVT::core::IO::GPIO::Direction::OUTPUT);
+    IO::GPIO &vicorIndicator_GPIO =
+            IO::getGPIO<APM::APMManager::Test_LED_1>(EVT::core::IO::GPIO::Direction::OUTPUT);
+
+    auto apmUart = APM::APMUart(&uart);
+
+    while (true) {
+        apmUart.printString("\n\rPlease enter a command\n\r");
+        apmUart.printString("Enter 'h' for help\n\r");
+        apmUart.printString(">> ");
+
+        apmUart.gets(APM::buf, APM::BUF_SIZE);
+        apmUart.printString("\n\r");
+
+        if (strncmp("h", APM::buf, APM::BUF_SIZE) == 0) {
```

Comment:
This is more of a question since it is just a test, but wouldn't it be more optimal to utilize a switch-case